src/components/ChatEntry.js

```diff
@@ -1,22 +1,30 @@
 import React from 'react';
 import './ChatEntry.css';
 import PropTypes from 'prop-types';
+import TimeStamp  from './TimeStamp';
 
 const ChatEntry = (props) => {
+
+  const heartColor = props.liked ? '❤️' : '🤍';
+  const bubbleDirection = props.id % 2 === 0 ? 'chat-entry remote' : 'chat-entry local';
+
   return (
-    <div className="chat-entry local">
-      <h2 className="entry-name">Replace with name of sender</h2>
+    <div className={bubbleDirection}>
+      <h2 className="entry-name">{props.sender}</h2>
       <section className="entry-bubble">
-        <p>Replace with body of ChatEntry</p>
-        <p className="entry-time">Replace with TimeStamp component</p>
-        <button className="like">🤍</button>
+        <p>{props.body}</p>
+        <p className="entry-time"><TimeStamp time={props.timeStamp}/></p>
+        <button className="like" onClick={() => props.toggleLike(props.id)}>{heartColor}</button>
       </section>
     </div>
   );
 };
 
 ChatEntry.propTypes = {
-  //Fill with correct proptypes
```

Comment:
Don't forget that you want to add **all** your `props` and give them a `propType`, you made use of `id` in here, so you would want to add that as well!